packages/shared/src/hooks/usePostMenuActions.ts

```diff
@@ -61,10 +64,20 @@ export const usePostMenuActions = ({
       SourcePermissions.PostDelete,
     );
 
+  const canPin = post?.source.currentMember?.permissions?.includes(
+    SourcePermissions.PostPin,
+  );
+
+  const { mutateAsync: onPinPost } = useMutation(
+    () => updatePinnedPost({ id: post.id, pinned: !post.pinnedAt }),
+    { onSuccess: onPinSuccessful },
```

Comment:
I would handle the mutation success here actually, the hook seems like the perfect place to have this logic.